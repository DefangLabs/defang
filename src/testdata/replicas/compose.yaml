services:
  default:
    image: nginx:latest
  replicated:
    image: nginx:latest
    deploy:
      replicas: 3
  autoscaled:
    image: nginx:latest
    x-defang-autoscaling: true
  deploy:
    image: nginx:latest
    deploy:
      resources:
        reservations:
          memory: 256M
          cpus: "0.25"
  managed:
    image: redis:latest
    x-defang-redis: true
