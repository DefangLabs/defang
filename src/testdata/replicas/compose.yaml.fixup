{
  "autoscaled": {
    "command": null,
    "entrypoint": null,
    "image": "nginx:latest",
    "networks": {
      "default": null
    }
  },
  "default": {
    "command": null,
    "entrypoint": null,
    "image": "nginx:latest",
    "networks": {
      "default": null
    }
  },
  "deploy": {
    "command": null,
    "deploy": {
      "resources": {
        "reservations": {
          "cpus": 0.25,
          "memory": "268435456"
        }
      },
      "placement": {}
    },
    "entrypoint": null,
    "image": "nginx:latest",
    "networks": {
      "default": null
    }
  },
  "managed": {
    "command": null,
    "entrypoint": null,
    "image": "redis:latest",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "host",
        "target": 6379,
        "protocol": "tcp"
      }
    ]
  },
  "replicated": {
    "command": null,
    "deploy": {
      "replicas": 3,
      "resources": {},
      "placement": {}
    },
    "entrypoint": null,
    "image": "nginx:latest",
    "networks": {
      "default": null
    }
  }
}