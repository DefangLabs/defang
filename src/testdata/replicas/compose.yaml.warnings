 ! service "autoscaled": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "default": high-availability mode requires at least 2 replicas or x-defang-autoscaling
 ! service "default": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "deploy": high-availability mode requires at least 2 replicas or x-defang-autoscaling
 ! service "managed": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "replicated": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
