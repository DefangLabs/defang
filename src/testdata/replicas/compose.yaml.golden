name: replicas
services:
  autoscaled:
    image: nginx:latest
    networks:
      default: null
    x-defang-autoscaling: true
  default:
    image: nginx:latest
    networks:
      default: null
  deploy:
    deploy:
      resources:
        reservations:
          cpus: 0.25
          memory: "268435456"
    image: nginx:latest
    networks:
      default: null
  managed:
    image: redis:latest
    networks:
      default: null
    x-defang-redis: true
  replicated:
    deploy:
      replicas: 3
    image: nginx:latest
    networks:
      default: null
networks:
  default:
    name: replicas_default
